tools/lint/clang_tidy.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+import argparse
+import sys
+import os
+import re
+import logging
+from utils import run_command, setup_logger
+
+def write_file(file_path: str, content: str):
+    """Write content to the specified file."""
+    with open(file_path, "w", encoding="utf-8") as f:
+        f.write(content)
+    logging.info("Wrote file: %s", file_path)
```

Comment:
I'd prefer if we did not have separate function for this. We are essentially writing
```py
write_file("file.txt", content)
```
instead of
```
with open(file_path, "w", encoding="utf-8") as f:
    f.write(content)
logging.info("Wrote file: %s", file_path)
```
, and that is excluding the function definition. This is not about number of lines, but readability. While I am fan of splitting into functions, so code gets easier to follow, here code just gets harder to follow.